faculty/clients/job.py

```diff
@@ -256,9 +387,46 @@ def list(self, project_id):
         endpoint = "/project/{}/job".format(project_id)
         return self._get(endpoint, JobSummarySchema(many=True))
 
-    def update_metadata(self, project_id, job_id, name, description):
+    def create(self, project_id, name, description, definition):
+        """Create a job.
+
+        Parameters
+        ----------
+        project_id : uuid.UUID
+        name: str
+        description: str
+        definition: Dict
```

Comment:
I more had in mind that users would pass a `JobDefinition` object, which we would then convert to the corresponding `dict` with the `JobDefinitionSchema`'s `dump` method. What do you think?